tests/test_datasets/test_internal_datasets.py

```diff
@@ -126,13 +126,97 @@ def test_not_present_part():
             pd.to_datetime("2022-01-01 00:00:00"),
             ("train", "test"),
         ),
+        (
+            "m3_monthly",
+            (144, 2856),
+            pd.to_datetime("2010-01-31 00:00:00"),
+            pd.to_datetime("2021-12-31 00:00:00"),
+            ("train", "test"),
+        ),
+        (
+            "m3_quarterly",
+            (72, 1512),
+            pd.to_datetime("2004-03-31 00:00:00"),
+            pd.to_datetime("2021-12-31 00:00:00"),
+            ("train", "test"),
+        ),
+        (
+            "m3_other",
+            (104, 348),
+            pd.to_datetime("1996-03-31 00:00:00"),
+            pd.to_datetime("2021-12-31 00:00:00"),
+            ("train", "test"),
+        ),
+        (
+            "m3_yearly",
+            (47, 1290),
+            pd.to_datetime("1975-12-31 00:00:00"),
+            pd.to_datetime("2021-12-31 00:00:00"),
+            ("train", "test"),
+        ),
     ],
 )
 def test_dataset_statistics(dataset_name, expected_shape, expected_min_date, expected_max_date, dataset_parts):
+    import ssl
+
+    ssl._create_default_https_context = ssl._create_unverified_context
     ts_full = load_dataset(dataset_name, parts="full")
     ts_parts = load_dataset(dataset_name, parts=dataset_parts)
     parts_rows = sum([ts.df.shape[0] for ts in ts_parts])
     assert ts_full.df.shape == expected_shape
     assert ts_full.index.min() == expected_min_date
     assert ts_full.index.max() == expected_max_date
     assert ts_full.df.shape[0] == parts_rows
+
+
+@pytest.mark.parametrize(
+    "dataset_name, expected_train_df_exog_shape, expected_test_df_exog_shape, expected_train_df_exog_dates, expected_test_df_exog_dates",
+    [
+        (
+            "m3_monthly",
+            (144, 1428),
+            (18, 1428),
+            (pd.to_datetime("2010-01-31 00:00:00"), pd.to_datetime("2021-12-31 00:00:00")),
+            (pd.to_datetime("2020-07-31 00:00:00"), pd.to_datetime("2021-12-31 00:00:00")),
+        ),
+        (
+            "m3_quarterly",
+            (72, 756),
+            (8, 756),
+            (pd.to_datetime("2004-03-31 00:00:00"), pd.to_datetime("2021-12-31 00:00:00")),
+            (pd.to_datetime("2020-03-31 00:00:00"), pd.to_datetime("2021-12-31 00:00:00")),
+        ),
+        (
+            "m3_other",
+            (104, 174),
+            (8, 174),
+            (pd.to_datetime("1996-03-31 00:00:00"), pd.to_datetime("2021-12-31 00:00:00")),
+            (pd.to_datetime("2020-03-31 00:00:00"), pd.to_datetime("2021-12-31 00:00:00")),
+        ),
+        (
+            "m3_yearly",
+            (47, 645),
+            (6, 645),
+            (pd.to_datetime("1975-12-31 00:00:00"), pd.to_datetime("2021-12-31 00:00:00")),
+            (pd.to_datetime("2016-12-31 00:00:00"), pd.to_datetime("2021-12-31 00:00:00")),
+        ),
+    ],
+)
+def test_m3_contains_df_exog(
```

Comment:
I'm not sure it is the best way to test this. May be we should make a test similar to `test_dataset_statistics` where we check parameters of exog.

Reply:
actually I check parameters of df_exog here

Reply:
Yes, but this test doesn't look like I can put other datasets here. Let's at least rename it to smth like `test_df_exog_statistics`